ckanext/dhis2harvester/templates/source/pivot_table_new_3.html

```diff
@@ -46,13 +46,72 @@ <h2>Configure Columns and Categories</h2>
+              <div class="row">
+                {% if target_type == 'other' %}
+                  {{ form.input("target_column_{}_{}".format(pivot_table.id, column.value), label=_('column'), value=column_data.target_column) }}
+                  {{ form.input("category_{};primary_category;{}".format(pivot_table.id, column.value), label=_('primary category'), value=column_data.categories['primary_category'] if column_data.categories else '') }}
+                  {{ form.input("category_{};secondary_category;{}".format(pivot_table.id, column.value), label=_('secondary category'), value=column_data.categories['secondary_category'] if column_data.categories else '') }}
+                {% else %}
+
+                  {% set column_options = [] %}
+                  {% for column in data.column_config[target_type].columns %}
+                    {% do column_options.append({'value': column, 'text': column}) %}
+                  {% endfor %}
+                  {{ form.select(
+                    'target_column;{};{}'.format(pivot_table.id, column.value),
+                    id='target_column;{};{}'.format(pivot_table.id, column.value),
+                    label='column',
+                    options=column_options,
+                    selected=column_data.target_column if column_data else column_options[0],
+                    error=errors.pivot_table_id,
+                    classes=['control-medium', 'col-xs-12', 'col-md-4']
+                    )
+                      }}
+
+                  {% for category, values in h.iteritems(data.column_config[target_type].categories) %}
+                    {% set category_options = [] %}
+                    {% for value in values %}
+                      {% do category_options.append({'value': value, 'text': value}) %}
+                    {% endfor %}
+                    {{ form.select(
+                      'category;{};{};{}'.format(pivot_table.id, category, column.value),
+                      id='category;{};{};{}'.format(pivot_table.id, category, column.value),
+                      label=category,
+                      options=category_options,
+                      selected=column_data.categories[category] if column_data else category_options[0].value,
+                      error=errors.pivot_table_id,
+                      classes=['control-medium', 'col-xs-12', 'col-md-4']
+                      )
+                        }}
+                  {% endfor %}
+                {% endif %}
+              </div>
+              <div class="row">
+                <div class="col-xs-12">
+                  {{ form.checkbox(
+                        'negative;{};{}'.format(pivot_table.id, column.value),
+                        label=_('Negative value'),
+                        checked=column_negative,
+                        error=errors.pivot_table_id
+                  )
+                      }}
+                </div>
+              </div>
+            </div>
+          </div>
```

Comment:
There's obviously a lot of code duplication that could be avoided here by making a Jinja2 macro or something. But if it works I suggest we just merge now and fix later?